public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -79,17 +79,24 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
       })
       .pipe(
         map(({ data }) => {
-          const results = data.results['annotationQuery'].tables[0].rows.map((v: any) => {
-            return {
-              annotation: annotation,
-              time: Date.parse(v[0]),
-              title: v[1],
-              tags: [],
-              text: v[3],
-            } as any;
+          const dataQueryResponse = toDataQueryResponse({
```

Comment:
this doesn't feel quite right...shouldn't DataSourceWithBackend take care of this for us @ryantxu?

Reply:
Yep, some input would be nice here, thanks! 👍

Reply:
I did something like this in my CloudWatch PR to minimize changes to the frontend. But not sure if this should be the approach long-term

```js
const frames = toDataQueryResponse({ data: data }).data as DataFrame[];
const table = toLegacyResponseData(frames[0]) as TableData;
return table.rows.map((v: any) => {
   return {
         annotation: annotation,
         time: Date.parse(v[0]),
         title: v[1],
         tags: [],
         text: v[3],
    } as any;
});
```

Reply:
Seems prettier for sure

Reply:
What you have seems fine, and will not change the front end at all.  However, you *may* want to use the automatic/out-of-the-box annotation support.  For that, you simply need to add:
https://github.com/grafana/timestream-datasource/blob/master/src/DataSource.ts#L28

If you want a custom query editor you can specify it like this:
https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/datasource.ts#L68